### PR TITLE
PR 16130 deprecate symbol ==> Symbol

### PR DESCRIPTION
**Pull request by jeffreysarnoff-dev**
Note: (from @hayd) this was eventually merged in #16154 (originally #15995).

PR 16130 deprecate symbol ==> Symbol (clean version)
[apologies for needing three PR numbers to do one thing]

### CODE REVIEW

**Comment by jeffreysarnoff-dev** [comment]
corrected one problem -- force pushed the change into the single commit.
how do I get AppVeyor and Travis to recheck?

**Comment by hayd** [comment]
It looks like you're yet to address all @stevengj's comments from #15995. In future it's better IMO to force push to the original branch / PR. That way those comments become become "outdated diffs" once they are fixed.

**Comment by jeffreysarnoff-dev** [comment]
@hayd I know you are correct.  I am very unfamiliar with this process and after trying it the 'right way' umpteen times without success I was able to proceed this less than perfect way.  Regarding @stevengj's comments, as there was an unexpected, indirect conflict that arose when I did attempt incorporating his comments -- one that I did not know how to fix, on the advice of others I went back to the original purpose which was to replace symbol with Symbol.  Once this is clean, the checks are good and it can be accepted, then it is possible to readdress the additional suggestions in a separate PR.

**Comment by jeffreysarnoff-dev** [comment]
The last change was to comment out a test in replutil.jl (done in accord with prior advice). The advice (as I understood it) was to eliminate the test involving Symbol generating: "expected failure but no error thrown" because this failure mode was no longer relevant.
[waiting for AppVeyor to see if I got the right one]

**Comment by jeffreysarnoff-dev** [comment]
The test commented out is at lines 230-2 of test/replutil.jl

**Comment by jeffreysarnoff-dev** [comment]
This began as [#15995](https://github.com/JuliaLang/julia/pull/15995) and took a detour into [#16308](https://github.com/JuliaLang/julia/pull/16038) before coming to rest here.
@nalimilan or someone else, if you confirm that the test I have commented out is the one to remove, I am happy to remove it and push that change back here.

**Comment by jeffreysarnoff-dev** [comment]
.. on second thought .. I will remove it and if that is wrong tell me (so you can see it in the edits)

File: test/replutil.jl [hunk 1]
```diff
@@ -227,8 +227,9 @@ let err_str,
     i = reinterpret(EightBitType, 0x54),
     j = reinterpret(EightBitTypeT{Int32}, 0x54)
 
-    err_str = @except_str Symbol() MethodError
-    @test contains(err_str, "MethodError: no method matching Symbol()")
+    #PR 16130 (Symbol)
+    #err_str = @except_str Symbol() MethodError
+    #@test contains(err_str, "MethodError: no method matching Symbol()")
```
Comment (tkelman) [inline test/replutil.jl]: how about using some other type that doesn't have a 0-argument constructor? that would capture the point of this test

Comment (Jeffrey-Sarnoff) [inline test/replutil.jl]: @tkelman I did not realize that here Symbol() is not important as Symbol per se. I will see how this works substituting Float64(), which generates a MethodError

File: test/replutil.jl [hunk 2]
```diff
@@ -227,8 +227,9 @@ let err_str,
     i = reinterpret(EightBitType, 0x54),
     j = reinterpret(EightBitTypeT{Int32}, 0x54)
 
-    err_str = @except_str Symbol() MethodError
-    @test contains(err_str, "MethodError: no method matching Symbol()")
+    #PR 16130, replaced Symbol() with Float64()
```
Comment (nalimilan) [inline test/replutil.jl]: This comment isn't useful here. We don't want to add a comment above each line to recall it's history (and git is there for this).

**Comment by nalimilan** [comment]
Apart from the comment this looks good to me!

**Comment by Jeffrey-Sarnoff** [comment]
@nalimilan  @tkelman  The only change was to remove the comment 
(both checks before that change worked).
"Build execution time has reached the maximum allowed time for your plan (120 minutes)."
Should I restart Appveyor? How do I do that?

**Comment by nalimilan** [comment]
That's a random failure, next time it will likely succeed.

**Comment by Jeffrey-Sarnoff** [comment]
red-clay-for-modified-cello-ensemble-score-image        
    [ _notes noted_](http://celloexpressions.com/music/piece/red-clay/)

**Comment by nalimilan** [comment]
Let's merge this?

**Comment by hayd** [comment]
Has `symbol` been deprecated or removed?

**Comment by JeffreySarnoff** [comment]
symbol has been deprecated

**Comment by jeffreysarnoff-dev** [comment]
@hayd OOPS! nice catch -- that got lost in translation -- remedying now

**Comment by jeffreysarnoff-dev** [comment]
@hayd symbol is once again deprecated, but now there are conflicts and this is where I have had trouble.  Each time I have tried to fix a merge conflict, I have messed up my repository.  
And I don't know what the conflict is; I see " Can’t automatically merge. Don’t worry, you can still create the pull request."
.

**Comment by nalimilan** [comment]
Damn, there are conflicts with master again. Could you rebase?

**Comment by jeffreysarnoff-dev** [comment]
@nalimilan I did rebase locally -- and I could push that up to github ... but each time I did that in the  past, there was trouble.  After pushing the correction to deprecated.jl, here is what I did locally:
git fetch
git rebase -i origin/master
[a git-rebase-todo file opened in an editor, it was empty and I left it so]

**Comment by jeffreysarnoff-dev** [comment]
@nalimilan git says "Successfully rebased and updated refs/heads/jas/deprecate_symbol"
locally

**Comment by KristofferC** [comment]
I don't think you want to interactively rebase. Just `git rebase master` should try to replay your PR on top of the lastest commit on the master branch. When a commit fails to merge git will yell at you and if you do `git status` you can see what files have a merge conflict. Fix these, add them with `git add` and then run `git rebase --continue`. Rinse and repeat.

**Comment by jeffreysarnoff-dev** [comment]
@KristofferC for me "master" is https://github.com/jeffreysarnoff-dev/julia which is now 6 commits behind JuliaLang:master.  my "upstream" is ..JuliaLang/julia.git.  And I have no more commits, I could edit the commit message in an inconsequential way if I need to be able to force push something that is local.
What should I do to get to the point that git rebase master does what you suggest?

**Comment by jeffreysarnoff-dev** [comment]
@KristofferC git status is telling me that (locally) I am on branch jas/deprecate_symbol and "Your branch is up-to-date with 'origin/jas/decprecate_symbol'

**Comment by nalimilan** [comment]
`git rebase origin/master` should use the "real" master, not the one from your clone.

**Comment by nalimilan** [comment]
Have a look at the GitHub docs to understand how all of this works. In particular:
https://help.github.com/articles/good-resources-for-learning-git-and-github/
https://help.github.com/articles/resolving-a-merge-conflict-from-the-command-line/

**Comment by jeffreysarnoff-dev** [comment]
@nalimilan I understand how git works .. I just don't like it.  I have been using an app, SmartGit, along with using command line git because the app prevents me from doing some things wrong. It may set me to do other things oddly.  It is that app that 'decided' origin is my github branch and upstream is JuliaLang.  My local master branch points up to my github fork.  Right now, I need use git on the command line to rebase things using JuliaLang master so I can see the conflicts and then address them.  Any guidance is welcome -- not about how git works, about how to play this current game of _cat's cradle_ to properly re-entangle things.

**Comment by jeffreysarnoff-dev** [comment]
When I close the git-assist App, it goes away and I can do anything I want from the command line.  Should I restart that app, it just picks up from whatever state the git repository is in right then.

**Comment by hayd** [comment]
Using `upstream` for JuliaLang is good, the only difference is you have to do `git rebase upstream/master`. You *may* also have to first explicitly `git fetch upstream`, depending on your git version/setup.

*Note: The conflict is just in the `deprecated.jl`.*

My advice is not to use a git app (or git within your IDE) until you have understood everything that the app/ide is doing at every step of the process.

**Comment by jeffreysarnoff-dev** [comment]
@hayd Thank you.

**Comment by jeffreysarnoff-dev** [comment]
red-clay-for-modified-cello-ensemble-score-image                                         
[above is from Red-Clay](http://celloexpressions.com/music/piece/red-clay/)

**Comment by JeffreySarnoff** [comment]
This has been a hike; at times I became weary and would have been uncomfortable without you:
@hayd @JeffBezanson  @Keno @KristofferC @nalimilan  @stevengj @timholy  @tkelman @vtjnash  @yuyichao (and any omitted by happenstance).

File: base/deprecated.jl [hunk 3]
```diff
@@ -368,6 +368,9 @@ end
 @deprecate flipud(A::AbstractArray) flipdim(A, 1)
 @deprecate fliplr(A::AbstractArray) flipdim(A, 2)
 
+@deprecate sub2ind{T<:Integer}(dims::Array{T}, sub::Array{T}) sub2ind(tuple(dims...), sub...)
```
Comment (nalimilan) [inline base/deprecated.jl]: You have a lot of unrelated changes in this file.

**Comment by jeffreysarnoff-dev** [comment]
can someone reattach this PR to 
https://github.com/jeffreysarnoff-dev/julia/tree/jas/deprecate_symbol
(which may be all fixed, but there is no rechecking happening because it got detached).

**Comment by jeffreysarnoff-dev** [comment]
sweeping away more bits --
